packages/flutter/lib/src/widgets/transitions.dart

```diff
@@ -142,6 +142,47 @@ class _AnimatedState extends State<AnimatedWidget> {
   Widget build(BuildContext context) => widget.build(context);
 }
 
+/// Convenience function for passing around a builder for a transiton's secondary animation.
+typedef DelegatedTransitionBuilder = Widget Function(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget? child);
+
+/// Placeholder
+@immutable
+class DelegatedTransition {
```

Comment:
This reminds me of [PageTransitionsBuilder](https://main-api.flutter.dev/flutter/material/PageTransitionsBuilder-class.html). I wonder if that class was created for a similar reason?

Reply:
It seems like it. It looks like it was used for comparing equality of `PageTransitionsTheme`.